pkg/foundation/grpcutil/websocket.go

```diff
@@ -112,6 +117,10 @@ func (p *webSocketProxy) proxy(w http.ResponseWriter, r *http.Request) {
 	ctx, cancelFn := context.WithCancel(r.Context())
 	defer cancelFn()
 	r = r.WithContext(ctx)
+	go func() {
+		<-p.done
```

Comment:
This goroutine will linger until Conduit is stopped, it would be nicer if we would also listen to `ctx.Done()` (the request context). To make this easier, we can use the utility methods in package `cchan`.